packages/core/src/readers/LlamaParseReader.ts

```diff
@@ -0,0 +1,110 @@
+import { Document } from "../Node";
+import { defaultFS } from "../env";
+import { GenericFileSystem } from "../storage/FileSystem";
+import { FileReader } from "./type";
+
+type ResultType = "text" | "markdown";
+
+/**
+ * Represents a reader for parsing files using the LlamaParse API.
+ * See https://github.com/run-llama/llama_parse
+ */
+export class LlamaParseReader implements FileReader {
+  // The API key for the LlamaParse API.
+  apiKey: string;
+  // The base URL of the Llama Parsing API.
+  baseUrl: string = "https://api.cloud.llamaindex.ai/api/parsing";
+  // The maximum timeout in seconds to wait for the parsing to finish.
+  maxTimeout = 2000;
+  // The interval in seconds to check if the parsing is done.
+  checkInterval = 1;
+  // Whether to print the progress of the parsing.
+  verbose = true;
+  resultType: ResultType = "text";
+
+  constructor(params: Partial<LlamaParseReader> = {}) {
+    Object.assign(this, params);
+    params.apiKey = params.apiKey ?? process.env.LLAMA_CLOUD_API_KEY;
+    if (!params.apiKey) {
+      throw new Error(
+        "API Key is required for LlamaParseReader. Please pass the apiKey parameter or set the LLAMA_CLOUD_API_KEY environment variable.",
+      );
+    }
+    this.apiKey = params.apiKey;
+  }
+
+  async loadData(
+    file: string,
+    fs: GenericFileSystem = defaultFS,
+  ): Promise<Document[]> {
+    if (!file.endsWith(".pdf")) {
+      throw new Error("Currently, only PDF files are supported.");
+    }
```

Comment:
No, use a file-type check instead. file ext is not file type guarantee

Reply:
@EmanuelCampos and @himself65 Okay, you guys convinced me :D ; mime-type check added. Currently, this makes more sense. My thinking was that LlamaParse will support multiple file types in the future. Then it makes sense that the client is just sending whatever file to the LlamaParse server, and LlamaParse is doing the mime-type check